src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -349,45 +390,66 @@ class OrderedRVD(
       partitioner.kType,
       newRangeBounds)
 
-    OrderedRVD(typ, newPartitioner, rdd.subsetPartitions(keep))
+    OrderedRVD(typ, newPartitioner, crdd.subsetPartitions(keep))
   }
 
-  def write(path: String, codecSpec: CodecSpec): Array[Long] = {
-    val (partFiles, partitionCounts) = rdd.writeRows(path, rowType, codecSpec)
-    val spec = OrderedRVDSpec(
+  override protected def rvdSpec(codecSpec: CodecSpec, partFiles: Array[String]): RVDSpec =
+    OrderedRVDSpec(
       typ,
       codecSpec,
       partFiles,
-      JSONAnnotationImpex.exportAnnotation(partitioner.rangeBounds, partitioner.rangeBoundsType))
-    spec.write(sparkContext.hadoopConfiguration, path)
-    partitionCounts
-  }
-
-  def zipPartitionsPreservesPartitioning[T: ClassTag](
+      JSONAnnotationImpex.exportAnnotation(
+        partitioner.rangeBounds,
+        partitioner.rangeBoundsType))
+
+  // FIXME: this is dangerous because the context is reused for left and
+  // right and will be reset based on when zipper's return value calls next
+  // but zipper might depend on one of the two children's next values not
+  // being erased
```

Comment:
That's the design.  This will always be a danger.  As we saw with flatMap, the correct thing to do is the producer that needs its value longer-lived than the consumer's lifetime needs to create and manage its own RegionValue.